kale/nbparser/parser.py

```diff
@@ -57,10 +61,11 @@ def parse_metadata(metadata):
             parsed_tags['step_names'] = [t]
             return parsed_tags
 
-        # now only `block` and `prev` tags remain to be parsed.
+        # now only `block|step` and `prev` tags remain to be parsed.
         tag_name, value = t.split(':')
         # name of the future Pipeline step
-        if tag_name == "block" and value:
+        # TODO: Deprecate `block` in future release
+        if tag_name in ["block", "step"] and value:
```

Comment:
The deprecation warning I'm referring to in my main comment could be inside this `if` codeblock.
I don't know if some deprecation library could work, but this should be sufficient:
```python
if tag_name == "block":
    log.warning(...)
```